### PR TITLE
Ruby : Enumerable implementation.

### PR DESCRIPTION
**Pull request by mailsg**
Hi,

This PR is raised to showcase the implementation of `MyList` class and `MyEnumerable` module with `Enumerable` methods. Below are the pointers covered : 
- Created `my_list.rb` file that defines the `MyList` class with an instance variable **@list**.
- Implemented the `each` method in `MyList` to yield successive members of **@list**, utilizing the functionality from the `MyEnumerable` module.
- Created `my_enumerable.rb` file that defines the `MyEnumerable` module.
- Implement `all`, `any`, and `filter` methods in **MyEnumerable**, which have the same functionality as their counterparts in the **Enumerable** module.

Kindly provide feedback on the same.

Thanks.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class MyList
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
```
Comment (Ismailco) [inline my_list.rb]: - [ ] You should create you're own `each` method instead of using the ruby method

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @mailsg, @techmoves, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (mailsg) [reply my_list.rb]: Done.

Comment (souad988) [reply my_list.rb]: Good job so far 👍🏼  , the idea here is to implement your own basic each method and make it execute a block on each element ==>

````ruby
def each
    return to_enum(:each) unless block_given?
    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
````

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @mailsg , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Use of modules and import them correctly
Good PR description

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_list.rb (hunk 1 above)
Comment (mailsg) [reply my_list.rb]: Implemented as instructed.

**Review by jerryowusu — APPROVED** [review]
Hi @mailsg 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

200w

## Highlights
Neat and well organized code :+1: 
Great job so far :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**